openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -710,6 +706,47 @@ Cache-Control: no-store
 }
 ```
 
+# Nonce Endpoint {#nonce-endpoint}
+
+This endpoint allows a Client to acquire a fresh `c_nonce` value without the overhead of a full Credential Request. A Credential Issuer that requires `c_nonce` values to be incorporated into proofs in the Credential Request (see (#credential-request)) MUST offer a Nonce Endpoint.
+
+The `nonce_endpoint` Credential Issuer Metadata parameter, as defined in (#credential-issuer-parameters), contains the URL of the Credential Issuer's Nonce Endpoint.
+
+
+## Nonce Request {#nonce-request}
+
+A request for a nonce is made by sending an HTTP GET request to the URL provided in the `nonce_endpoint` Credential Issuer Metadata parameter.
+
+Below is a non-normative example of a Nonce Request:
+
+```
+GET /nonce HTTP/1.1
+Host: credential-issuer.example.com
+```
+
+## Nonce Response {#nonce-response}
+
+The Credential Issuer provides a nonce value in the HTTP response with a 2xx status code and the following parameters included as top-level members in the message body of the HTTP response using the application/json media type:
+
+* `c_nonce`: REQUIRED. String containing a nonce to be used when creating a proof of possession of the key proof (see (#credential-request)).
```

Comment:
I suggest to rename c_nonce to nonce, making the nonce endpoint of general purpose. A nonce is issued by an entity for a specific scope known to its issuer, we don't need to use a particular member name to say that it is a credential nonce, it's a only nonce

nonce and nonce_expires_in would be enough from my perspective, allowing the nonce endpoint to be a general and reusable endpoint within the OAuth framework

Reply:
A general and reusable endpoint within the OAuth framework was in no way a goal of this work. Not at all. 

And I used c_nonce (a name that I honestly think is terrible for many reasons) to allow for this PR to fit into the broader document more easily.

Reply:
Okay, since this Nonce Endpoint is specifically tailored for credential issuance within the sole scope of the OpenID4VCI specs, it would make sense to rename the endpoint metadata parameter from `nonce_endpoint` to `credential_nonce_endpoint`.

In this way, other nonce endpoints created for other OpenID/IETF specs will not collide with this one.
The related section should therefore be renamed to "Credential Nonce Endpoint," adding "Credential" as a prefix to contextualize its scope.

Reply:
As @jogu points out (thanks @jogu!) there is no problem with collision because this endpoint is at the credential issuer. 

This seems to be in the same vein of what I said about this in slack to you @peppelinux yesterday but bears repenting, "the context is totally different."

Reply:
bo-ring :-)

Reply:
I menat IANA metadata registry

https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml

OAuth Authorization Server Metadata

nonce_endpoint might collide over there, yes different contexts (openid != IETF) but if this endpoint is specialized for credential issuance it would be better to give to it a name pointing exactly and esplicitly in this way

Reply:
that context is still totally different

Reply:
after a maybe productive but definitely not appropriate side discussion about this - I think we can agree to disagree

Reply:
> I menat IANA metadata registry
> 
> https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml
> 
> OAuth Authorization Server Metadata
> 
> nonce_endpoint might collide over there, yes different contexts (openid != IETF) but if this endpoint is specialized for credential issuance it would be better to give to it a name pointing exactly and esplicitly in this way

Just to try to clarify, the nonce endpoint here is part of the issuer metadata. We will not be making an IANA registration for nonce_endpoint in the OAuth Authorization Server Metadata because we're not defining a nonce endpoint for the authorization server. The suggested nonce_endpoint is (technically) a resource server endpoint as that's how we define the credential issuer.

Adding nonce_endpoint here would not prevent the OAuth WG defining a nonce endpoint for authorization servers, nor would it conflict with any such endpoint. It also wouldn't stop the OAuth WG defining a general nonce endpoint for resource servers should they choose to do so, as resource server metadata would also not have a name space that overlaps with credential issuer metadata (i.e. you could have a nonce endpoint defined in `.well-known/oauth-resource server` and a potentially different nonce endpoint defined in `/.well-known/openid-credential-issuer`.